libsel4camkes/include/camkes/virtqueue.h

```diff
@@ -147,7 +147,7 @@ void *camkes_virtqueue_driver_offset_to_buffer(virtqueue_driver_t *virtqueue, ui
  * @param size the size of the buffer
  * @return 0 on success, -1 on fail
  */
-int camkes_virtqueue_driver_send_buffer(virtqueue_driver_t *vq, void *buffer, size_t size);
+int camkes_virtqueue_driver_send_buffer(virtqueue_driver_t *vq, const void *buffer, size_t size);
```

Comment:
I think this function is also used for enqueuing buffers in the virtqueue to receive data on? 

In this case send means that the buffer is being sent to the other end of the queue (the device side). If the queue is used for transmitting data then the data in the buffer would be read by the other end, but if the queue is being used for receiving data then the other side would modify the contents before sending it back.

Reply:
Fair point. Would it make sense to separate the APIs then for the different use cases to support proper semantics?

Reply:
What if the semantics are just enqueuing into available hands a mutable reference over, and receiving from the other queue takes it back. Whether the buffers are written to by the other side is a higher level concern depending on the protocol using the buffers.